init.lua

```diff
@@ -119,12 +126,18 @@ end
 
 -- Register punchplayer callback.
 minetest.register_on_punchplayer(function(player, hitter, time_from_last_punch, tool_capabilities, dir, damage)
+
+	-- check if hitter != Player 
+	if hitter:is_player() == false then
+		return monster_do_no_damage 
```

Comment:
No. Check the state of `monster_damage`, and then return either `KILL_OK` or `KILL_NO`, it is ambiguous what will happen here like this.

Reply:
i my experiment the rerurn "false" means no damage and "true" damage to the player 
i do some changes for a better reading of the code